### PR TITLE
Triangulate backward compatibility check commands

### PR DESCRIPTION
**Pull request by yogeshnikam671**
**What**:
Created an abstract class which has the common logic around backward compatibility check. This helps reuse the logic in other protocols easily.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Tested locally
- [ ] Sonar Quality Gate

### CODE REVIEW

File: application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+const val ONE_INDENT = "  "
+const val TWO_INDENTS = "${ONE_INDENT}${ONE_INDENT}"
+
+@Component
+@Command(
+    name = "backwardCompatibilityCheck",
+    mixinStandardHelpOptions = true,
+    description = [
+        "Checks backward compatibility of a directory across the current HEAD and the main branch",
+        "DEPRECATED: This command will be removed in the next major release. Use 'backward-compatibility-check' command instead."
```
Comment (yogeshnikam671) [inline application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt]: Let's put a date or the release number when we will be completely deprecating/removing this.

File: application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt [hunk 2]
```diff
@@ -149,7 +157,11 @@ class BackwardCompatibilityCheckCommand(
                     // newer => the file with changes on the branch
                     val (newer, unusedExamples) = OpenApiSpecification.fromFile(specFilePath).toFeature().loadExternalisedExamplesAndListUnloadableExamples()
 
-                    val olderFile = gitCommand.getFileInTheDefaultBranch(specFilePath, treeishWithChanges)
+                    val olderFile = gitCommand.getFileInTheBaseBranch(
```
Comment (yogeshnikam671) [inline application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt]: Can be called getFileInBranch. Default the last parameter to default branch.

File: application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+        names = ["--base-branch"],
+        description = [
+            "Base branch to compare the changes against",
+            "Default value is the local origin HEAD of the current branch"
+        ],
+        required = false
+    )
+    var baseBranch: String? = null
+
+    @Option(names = ["--target-path"], description = ["Specification file or folder to run the check against"], required = false)
```
Comment (yogeshnikam671) [inline application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt]: Needs better description.

File: application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt (hunk 2 above)
Comment (yogeshnikam671) [reply application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt]: Addressed

File: application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt (hunk 3 above)
Comment (yogeshnikam671) [reply application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt]: Addressed

**Review by harikrishnan83 — APPROVED** [review]
Looks good to me

**Pull request merged**